packages/template-retail-react-app/app/pages/account/index.test.js

```diff
@@ -102,80 +135,87 @@ test('Provides navigation for subpages', async () => {
     )
 })
 
-test('Renders account detail page by default for logged-in customer', async () => {
-    renderWithProviders(<MockedComponent />)
-    expect(await screen.findByTestId('account-page')).toBeInTheDocument()
-    expect(await screen.findByTestId('account-detail-page')).toBeInTheDocument()
-    expect(screen.getByText('Testing Tester')).toBeInTheDocument()
-    expect(screen.getByText('customer@test.com')).toBeInTheDocument()
-    expect(screen.getByText('(727) 555-1234')).toBeInTheDocument()
-})
+describe('Render and logs out', function () {
+    test('Renders account detail page by default for logged-in customer, and can log out', async () => {
+        renderWithProviders(<MockedComponent />)
+
+        await waitFor(() => expect(window.location.pathname).toEqual(`${expectedBasePath}/login`))
+        // Render user profile [age
+        await waitFor(() => {
+            expect(window.location.pathname).toEqual(`${expectedBasePath}/account`)
+            expect(screen.getByTestId('account-detail-page')).toBeInTheDocument()
+            expect(screen.getByText('Testing Tester')).toBeInTheDocument()
+            expect(screen.getByText('customer@test.com')).toBeInTheDocument()
+            expect(screen.getByText('(727) 555-1234')).toBeInTheDocument()
+        })
 
-test('Allows customer to sign out', async () => {
-    renderWithProviders(<MockedComponent />, {
-        wrapperProps: {siteAlias: 'uk', appConfig: mockConfig.app}
-    })
-    expect(await screen.findByTestId('account-detail-page')).toBeInTheDocument()
-    user.click(screen.getAllByText(/Log Out/)[0])
-    await waitFor(() => {
-        expect(window.location.pathname).toEqual(`${expectedBasePath}/login`)
+        user.click(screen.getAllByText(/Log Out/)[0])
+        await waitFor(() => {
+            expect(window.location.pathname).toEqual(`${expectedBasePath}/login`)
+        })
     })
```

Comment:
nit, this feels like two unit tests

1. renders component and assert for data on the screen
2. click log out and assert path changes

Reply:
smaller unit tests are easier to maintain

Reply:
In this case, it is not necessary since in order to get to do the logout, we have to make sure that the page has been rendered, meaning we will have to await/assert the following block again before logging out. It is sorta duplicated, so I put them together. 
```
// Render user profile [age
        await waitFor(() => {
            expect(window.location.pathname).toEqual(`${expectedBasePath}/account`)
            expect(screen.getByTestId('account-detail-page')).toBeInTheDocument()
            expect(screen.getByText('Testing Tester')).toBeInTheDocument()
            expect(screen.getByText('customer@test.com')).toBeInTheDocument()
            expect(screen.getByText('(727) 555-1234')).toBeInTheDocument()
        })
```